### PR TITLE
Add validation to post model

### PR DESCRIPTION
**Pull request by binyamolango**
# Hi Reviewer 👋

Here are the changes I implemented to the project

1. Added presence validations for the name in the User model
2. Added numericality only_integer and greater_than_or_equal_to validations for the PostsCounter in the User model
3. Added presence validations for the title in the Post model
4. Added length validations for the title in the Post model
5. Added numericality only_integer and greater_than_or_equal_to validations for the CommentsCounter in the Post model
6. Added numericality only_integer and greater_than_or_equal_to validations for the LikesCounter in the Post model
7. Added unit specs for all of my models' methods and validations

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@ class User < ActiveRecord::Base
   has_many :likes
   has_many :comments
 
+  validates :name, presence: true
+  validates :posts_count, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   def recent_posts
```
Comment (danmainah) [inline app/models/user.rb:9]: - [ ] @binyamolango Great work adding all the validation for post and user models, however, you forgot to add tests for all of your models' methods and validations. Kindly consider adding these tests and making sure they are all passing.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @binyamolango , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added validations for the user model correctly as per the milestone requirement.
✔️ Added validations for post model correctly as per the milestone requirement.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @binyamolango👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- Good job with validations. ✔️
- Good job with model specs. ✔️ 
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️

- [ ] Great job adding model tests as requested by the previous reviewer but I can't find the `db` directory in the project (you removed it from the project) so not able to run the tests. Please make sure to add back the `db` directory and all tests should pass without any errors.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by IngDavidTM — APPROVED** [review]
Hi @binyamolango 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**